### PR TITLE
[Net] Turn net structures into dumb storage classes

### PR DESCRIPTION
**Pull request by Fuzzbawls**
Backport of https://github.com/bitcoin/bitcoin/pull/8128 as part of #1374. Based on top of #1640

Original Description:

> CNetAddr/CService/CSubNet lose their string constructors, they must now have lookup operations performed externally. This means that functions/classes that depend on them are no longer dependent on any particular lookup mechanism.
> 
> From a high level: New resolvers/parsers may now be used for net operations. libbtcnet will replace what remains of netbase with async versions.

Additional work has been done in the following commits:
* fcef585 - Removes a useless RPC command
* 235c33e - Refactors the masternode.conf parsing with regards to valid ports
* 3ddd35e - Adds additional sanity checks during client init for MN port numbers and listen options.

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -142,8 +142,9 @@ BITCOIN_CORE_H = \
   messagesigner.h \
   miner.h \
   mruset.h \
-  netbase.h \
   net.h \
+  netaddresses.h \
```
Comment (random-zebra) [inline src/Makefile.am:146]: `netaddress.h` ?

Comment (Fuzzbawls) [reply src/Makefile.am]: fixed

**Comment by Fuzzbawls** [comment]
rebased now that #1640 has been merged

File: src/masternode.cpp [hunk 2]
```diff
@@ -489,14 +497,13 @@ bool CMasternodeBroadcast::CheckSignature() const
     return true;
 }
 
-bool CMasternodeBroadcast::CheckDefaultPort(std::string strService, std::string& strErrorRet, std::string strContext)
+bool CMasternodeBroadcast::CheckDefaultPort(CService service, std::string& strErrorRet, const std::string& strContext)
```
Comment (furszy) [inline src/masternode.cpp:500]: Would be good to pass the service by reference

File: src/netbase.cpp [hunk 3]
```diff
@@ -1236,55 +1236,85 @@ CSubNet::CSubNet() : valid(false)
...
+    std::vector<CNetAddr> vIP;
+
     std::string strAddress = strSubnet.substr(0, slash);
     if (LookupHost(strAddress.c_str(), vIP, 1, false))
     {
-        network = vIP[0];
-        if (slash != strSubnet.npos) {
+        CNetAddr network = vIP[0];
+        if (slash != strSubnet.npos)
+        {
```
Comment (random-zebra) [inline src/netbase.cpp:683]: styling nit: no need for this change- we usually put branches on the same line with `if`s

File: src/net.cpp [hunk 4]
```diff
@@ -1824,8 +1826,11 @@ void StartNode(boost::thread_group& threadGroup, CScheduler& scheduler)
         semOutbound = new CSemaphore(nMaxOutbound);
     }
 
-    if (pnodeLocalHost == NULL)
-        pnodeLocalHost = new CNode(INVALID_SOCKET, CAddress(CService("127.0.0.1", 0), nLocalServices));
+    if (pnodeLocalHost == NULL) {
```
Comment (random-zebra) [inline src/net.cpp:1828]: styling nit: `nullptr`

File: src/netbase.cpp [hunk 5]
```diff
@@ -1236,55 +1236,85 @@ CSubNet::CSubNet() : valid(false)
...
+                if (LookupHost(strNetmask.c_str(), vIP, 1, false)) {
+                    ret = CSubNet(network, vIP[0]);
+                    return ret.IsValid();
                 }
             }
         }
-    } else {
-        valid = false;
+        else
+        {
```
Comment (random-zebra) [inline src/netbase.cpp]: styling nit: `} else {`

**Pull request merged**